service/history/ndc/workflow_resetter.go

```diff
@@ -349,23 +392,54 @@ func (r *workflowResetterImpl) persistToDB(
 		return nil
 	}
 
-	currentMutableState := currentWorkflow.GetMutableState()
-	currentRunID := currentMutableState.GetExecutionState().GetRunId()
-	currentCloseVersion, err := currentMutableState.GetCloseVersion()
+	// We have 3 different runs to update here - the base run, current run & the new run. But current is not yet closed.
+	if currentWorkflowMutation == nil {
+		return currentWorkflow.GetContext().ConflictResolveWorkflowExecution(
+			ctx,
+			r.shardContext,
+			persistence.ConflictResolveWorkflowModeUpdateCurrent,
+			baseWorkflow.GetMutableState(),
+			resetWorkflow.GetContext(),
+			resetWorkflow.GetMutableState(),
+			currentWorkflow.GetContext(),
+			currentWorkflow.GetMutableState(),
+			workflow.TransactionPolicyActive,
+			workflow.TransactionPolicyActive.Ptr(),
+			workflow.TransactionPolicyActive.Ptr(),
+		)
+	}
+
+	// we are here because currentMutation is closed. So we just need to close base and the new run and perform the transaction.
+	resetWorkflowSnapshot, resetWorkflowEventsSeq, err := resetWorkflow.GetMutableState().CloseTransactionAsSnapshot(
+		workflow.TransactionPolicyActive,
+	)
 	if err != nil {
 		return err
 	}
-
-	return resetWorkflow.GetContext().CreateWorkflowExecution(
+	baseSnapshot, baseEventsSeq, err := baseWorkflow.GetMutableState().CloseTransactionAsSnapshot(
+		workflow.TransactionPolicyActive,
+	)
+	if err != nil {
+		return err
+	}
+	resetRunVersion := resetWorkflow.GetMutableState().GetCurrentVersion()
+	currentRunVerson := currentWorkflow.GetMutableState().GetCurrentVersion()
+	if _, _, _, err := r.transaction.ConflictResolveWorkflowExecution(
```

Comment:
plz add some comments re why we are calling this method which is meant for xdc conflict resolution.

Reply:
Done.